### PR TITLE
Allow requestFullscreen() to be called from an orientation change event (closes #34)

### LINKED ISSUE #34: Allow fullscreen to be triggered by an orientation change event
On mobile devices, a common user experience is to have videos play inline when the device is in portrait mode and fullscreen when the user changes the orientation to landscape. Unfortunately, this is not possible on the Web because of the user gesture requirement to go fullscreen.

Changing the user gesture definition to include orientation change is unlikely a good idea because it wouldn't be appropriate for more dangerous features than fullscreen (eg. gUM). However, for fullscreen, and maybe other features, orientation change can be a good signal.

Chrome is willing to ship this ASAP. Anne, would Mozilla be interested to implement this change?

/CC @foolip

### PR DESCRIPTION
**Pull request by mounirlamouri**
This is using https://github.com/w3c/screen-orientation/pull/90

### CODE REVIEW

**Comment by mounirlamouri** [comment]
@foolip PTAL

**Comment by foolip** [comment]
@annevk should review this. I see that Fullscreen hasn't adopted the "triggered by user activation" rename yet, with that and a similarly named concept in the Screen Orientation spec this would look pretty clean I think.

I'll keep noting at each step that I don't think it's a good idea to have an API that cannot be implemented without bad frames. Before this is implemented I think that some form of sign-off from the relevant people at each browser engine would be good, and if none think that this matters then I will be quiet. (Shipping it and having them complain about exactly this after the fact would be sad.)

**Comment by mounirlamouri** [comment]
For the record, Mozilla stated publicly that they were okay with the bad frame. I sent an email to Microsoft and Apple (@foolip was CC'd). Microsoft said that they were okay with the feature and with the bad frame. Apple made no comment. All the browsers implementing the Screen Orientation API and the Fullscreen API on mobile are supportive of this change.

**Comment by mounirlamouri** [comment]
@foolip @annevk  PTAL. I simplified the logic living in the fullscreen spec.

**Comment by annevk** [comment]
This looks okay. I'd prefer if this pointed at the new term in HTML and has a reference to Screen Orientation API.

**Comment by annevk** [comment]
@foolip could you point to a canonical explanation of the bad frame problem? Might be useful in the future and we should maybe include it in the commit message.

**Comment by mounirlamouri** [comment]
https://github.com/whatwg/xref/pull/12 is doing the changes required in xref.

**Comment by mounirlamouri** [comment]
@annevk PTAL, your comments are applied. I can update the commit message if @foolip wants.

**Comment by foolip** [comment]
I picked up the term "bad frame" from the [Intent to Deprecate: overflowchanged event](https://groups.google.com/a/chromium.org/d/msg/blink-dev/1RCzsyEyNU8/gs7gLOjIU78J) on blink-dev two years ago, I don't know if this is what it's usually called.

But let me illustrate it with commented code:

```JavaScript
screen.orientation.addEventListener('change', function() {
  // called in the first animation frame task after the user rotates
  // the viewport width and height have changed and there will be a layout before the next paint
  someElement.requestFullscreen();
  document.addEventListener('fullscreenchange', function() {
    // called in the first animation frame task after going fullscreen
    // the viewport size (most likely height) has changed again
    // there will be a layout before the next paint
    // probably some things will move vertically
  });
});

File: Overview.html [hunk 1]
```diff
@@ -757,6 +765,7 @@ <h2 class="no-num" id="acknowledgments">Acknowledgments</h2>
 João Eiras,
 Josh Soref,
 Matt Falkenhagen,
+Mounir Lamouri,
```
Comment (annevk) [inline Overview.html]: Mo > Mi.

File: Overview.src.html [hunk 2]
```diff
@@ -180,6 +180,15 @@ <h2 class="no-num no-toc">Table of Contents</h2>
...
 <p><dfn>Fullscreen is supported</dfn> if there is no previously-established user
 preference, security risk, or platform limitation.
+
+<p>An algorithm is <dfn>allowed to request fullscreen</dfn> if one of the following conditions are
+true:
+
+ <ul>
+  <li>The algorithm is <span data-anolis-spec=html>triggered by user activation</span>.
+
+  <li>The algorithm is <span data-anolis-spec=screen-orientation>triggered by a user generated orientation change</span>.
```
Comment (annevk) [inline Overview.src.html]: Newline before `<span>` and please use `<li><p>` here too.

File: Overview.src.html [hunk 3]
```diff
@@ -180,6 +180,15 @@ <h2 class="no-num no-toc">Table of Contents</h2>
 
 <p><dfn>Fullscreen is supported</dfn> if there is no previously-established user
 preference, security risk, or platform limitation.
+
+<p>An algorithm is <dfn>allowed to request fullscreen</dfn> if one of the following conditions are
+true:
```
Comment (annevk) [inline Overview.src.html]: "if one of the following is true"

Also this might need to be wrapped per 100column rule.

File: Overview.html (hunk 1 above)
Comment (mounirlamouri) [inline Overview.html]: One day I will know my alphabet :)

File: Overview.src.html (hunk 2 above)
Comment (mounirlamouri) [inline Overview.src.html]: Done.

File: Overview.src.html (hunk 3 above)
Comment (mounirlamouri) [inline Overview.src.html]: Done

**Comment by mounirlamouri** [comment]
@annevk PTAL. I've applied all the comments.

**Pull request merged**